coreax/kernel.py

```diff
@@ -348,6 +348,45 @@ def calculate_kernel_matrix_row_sum_mean(
         return self.calculate_kernel_matrix_row_sum(x, max_size) / (1.0 * x.shape[0])
 
 
+class LinearKernel(Kernel):
+    """Define a linear kernel."""
+
+    length_scale: float = 1.0
+    output_scale: float = 1.0
+
+    @override
+    def compute_elementwise(
+        self,
+        x: ArrayLike,
+        y: ArrayLike,
+    ) -> Array:
+        return x.dot(y)
```

Comment:
Pyright rightly raises and errpr here because the `ArrayLike` type hint is very permissive. For example, if x is an integer, it won't have a `x.dot` attribute. Instead you can use  `jnp.dot(x, y)`.

Reply:
Fixed